File: mobile/ios/INTcoinTheme.swift
// Copyright (c) 2024-2026 The INTcoin Core developers
// Distributed under the MIT software license

import SwiftUI

/// INTcoin brand colors - dark theme matching Qt desktop wallet
/// WCAG 2.1 AA compliant with proper contrast ratios
struct INTcoinColors {
    // MARK: - Primary Backgrounds
    static let background = Color(hex: "0a0f1e")           // Deep navy - main background
    static let backgroundSecondary = Color(hex: "141b2e")  // Slightly lighter - cards
    static let backgroundTertiary = Color(hex: "1e2842")   // Input fields, buttons

    // MARK: - Accent Colors
    static let primary = Color(hex: "3399ff")              // INTcoin blue - primary actions
    static let primaryHover = Color(hex: "66b3ff")         // Blue hover state
    static let accent = Color(hex: "00d4ff")               // Cyan - highlights, selected states

    // MARK: - Semantic Colors
    static let success = Color(hex: "10b75f")              // Green - received, confirmed
    static let warning = Color(hex: "f59e0b")              // Amber - pending, unconfirmed
    static let error = Color(hex: "ef4444")                // Red - errors, failed, send

    // MARK: - Text Colors
    static let textPrimary = Color(hex: "f8fafc")          // White - primary text
    static let textSecondary = Color(hex: "94a3b8")        // Gray - secondary text
    static let textTertiary = Color(hex: "64748b")         // Darker gray - hints

    // MARK: - Border Colors
    static let border = Color(hex: "334155")               // Border, dividers
    static let borderFocused = Color(hex: "3399ff")        // Focus ring

    // MARK: - Gradients
    static let gradientPrimary = LinearGradient(
        colors: [Color(hex: "3399ff"), Color(hex: "00d4ff")],
        startPoint: .leading,
        endPoint: .trailing
    )

    static let gradientBackground = LinearGradient(
        colors: [Color(hex: "0a0f1e"), Color(hex: "141b2e")],
        startPoint: .top,
        endPoint: .bottom
    )
}

/// INTcoin typography styles
struct INTcoinTypography {
    static let largeTitle = Font.system(size: 34, weight: .bold, design: .rounded)
    static let title = Font.system(size: 28, weight: .bold, design: .rounded)
    static let title2 = Font.system(size: 22, weight: .semibold, design: .rounded)
    static let headline = Font.system(size: 17, weight: .semibold)
    static let body = Font.system(size: 17)
    static let callout = Font.system(size: 16)
    static let subheadline = Font.system(size: 15)
    static let footnote = Font.system(size: 13)
    static let caption = Font.system(size: 12)
    static let monospace = Font.system(size: 14, weight: .regular, design: .monospaced)
    static let balance = Font.system(size: 42, weight: .bold, design: .rounded)
}

/// INTcoin design constants
struct INTcoinDesign {
    static let cornerRadiusSmall: CGFloat = 8
    static let cornerRadiusMedium: CGFloat = 12
    static let cornerRadiusLarge: CGFloat = 16
    static let cornerRadiusXL: CGFloat = 24

    static let paddingSmall: CGFloat = 8
    static let paddingMedium: CGFloat = 16
    static let paddingLarge: CGFloat = 24
    static let paddingXL: CGFloat = 32

    static let minTouchTarget: CGFloat = 44
    static let buttonHeight: CGFloat = 52
    static let iconSizeMedium: CGFloat = 24
    static let iconSizeLarge: CGFloat = 32
    static let iconSizeXL: CGFloat = 48
}

// MARK: - Color Extension

extension Color {
    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - View Modifiers

struct INTcoinCardStyle: ViewModifier {
    func body(content: Content) -> some View {
        content
            .background(INTcoinColors.backgroundSecondary)
            .cornerRadius(INTcoinDesign.cornerRadiusMedium)
            .overlay(
                RoundedRectangle(cornerRadius: INTcoinDesign.cornerRadiusMedium)
                    .stroke(INTcoinColors.border, lineWidth: 1)
            )
    }
}

struct INTcoinPrimaryButtonStyle: ButtonStyle {
    var isEnabled: Bool = true

    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(INTcoinTypography.headline)
            .foregroundColor(isEnabled ? INTcoinColors.background : INTcoinColors.textTertiary)
            .frame(maxWidth: .infinity)
            .frame(height: INTcoinDesign.buttonHeight)
            .background(isEnabled ?
                (configuration.isPressed ? INTcoinColors.primaryHover : INTcoinColors.primary) :
                INTcoinColors.backgroundTertiary
            )
            .cornerRadius(INTcoinDesign.cornerRadiusSmall)
    }
}

struct INTcoinSecondaryButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .font(INTcoinTypography.headline)
            .foregroundColor(INTcoinColors.primary)
            .frame(maxWidth: .infinity)
            .frame(height: INTcoinDesign.buttonHeight)
            .background(configuration.isPressed ?
                INTcoinColors.primary.opacity(0.2) :
                INTcoinColors.backgroundTertiary
            )
            .cornerRadius(INTcoinDesign.cornerRadiusSmall)
            .overlay(
                RoundedRectangle(cornerRadius: INTcoinDesign.cornerRadiusSmall)
                    .stroke(INTcoinColors.primary, lineWidth: 2)
            )
    }
}

struct INTcoinTextFieldStyle: ViewModifier {
    var isFocused: Bool = false

    func body(content: Content) -> some View {
        content
            .padding(INTcoinDesign.paddingMedium)
            .background(INTcoinColors.backgroundSecondary)
            .foregroundColor(INTcoinColors.textPrimary)
            .cornerRadius(INTcoinDesign.cornerRadiusSmall)
            .overlay(
                RoundedRectangle(cornerRadius: INTcoinDesign.cornerRadiusSmall)
                    .stroke(isFocused ? INTcoinColors.borderFocused : INTcoinColors.border, lineWidth: 2)
            )
    }
}

// MARK: - View Extensions

extension View {
    func intcoinCard() -> some View {
        modifier(INTcoinCardStyle())
    }

    func intcoinTextField(isFocused: Bool = false) -> some View {
        modifier(INTcoinTextFieldStyle(isFocused: isFocused))
    }
}

File: mobile/android/src/main/java/org/intcoin/wallet/ui/theme/INTcoinTheme.kt
// Copyright (c) 2024-2026 The INTcoin Core developers
// Distributed under the MIT software license

package org.intcoin.wallet.ui.theme

import android.app.Activity
import androidx.compose.foundation.isSystemInDarkTheme
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.SideEffect
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.toArgb
import androidx.compose.ui.platform.LocalView
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat

/**
 * INTcoin brand colors - dark theme matching Qt desktop wallet
 * WCAG 2.1 AA compliant with proper contrast ratios
 */
object INTcoinColors {
    // Primary Backgrounds
    val Background = Color(0xFF0A0F1E)           // Deep navy - main background
    val BackgroundSecondary = Color(0xFF141B2E)  // Slightly lighter - cards
    val BackgroundTertiary = Color(0xFF1E2842)   // Input fields, buttons
    val Surface = Color(0xFF141B2E)
    val SurfaceVariant = Color(0xFF1E2842)

    // Accent Colors
    val Primary = Color(0xFF3399FF)              // INTcoin blue - primary actions
    val PrimaryHover = Color(0xFF66B3FF)         // Blue hover state
    val Accent = Color(0xFF00D4FF)               // Cyan - highlights, selected states
    val Secondary = Color(0xFF00D4FF)

    // Semantic Colors
    val Success = Color(0xFF10B75F)              // Green - received, confirmed
    val Warning = Color(0xFFF59E0B)              // Amber - pending, unconfirmed
    val Error = Color(0xFFEF4444)                // Red - errors, failed

    // Text Colors
    val TextPrimary = Color(0xFFF8FAFC)          // White - primary text
    val TextSecondary = Color(0xFF94A3B8)        // Gray - secondary text
    val TextTertiary = Color(0xFF64748B)         // Darker gray - hints
    val OnPrimary = Color(0xFF0A0F1E)            // Text on primary color

    // Border Colors
    val Border = Color(0xFF334155)               // Border, dividers
    val BorderFocused = Color(0xFF3399FF)        // Focus ring
}

/**
 * INTcoin dark color scheme for Material 3
 */
private val INTcoinDarkColorScheme = darkColorScheme(
    primary = INTcoinColors.Primary,
    onPrimary = INTcoinColors.OnPrimary,
    primaryContainer = INTcoinColors.Primary.copy(alpha = 0.2f),
    onPrimaryContainer = INTcoinColors.Primary,
    secondary = INTcoinColors.Secondary,
    onSecondary = INTcoinColors.OnPrimary,
    secondaryContainer = INTcoinColors.Secondary.copy(alpha = 0.2f),
    onSecondaryContainer = INTcoinColors.Secondary,
    tertiary = INTcoinColors.Accent,
    onTertiary = INTcoinColors.OnPrimary,
    background = INTcoinColors.Background,
    onBackground = INTcoinColors.TextPrimary,
    surface = INTcoinColors.Surface,
    onSurface = INTcoinColors.TextPrimary,
    surfaceVariant = INTcoinColors.SurfaceVariant,
    onSurfaceVariant = INTcoinColors.TextSecondary,
    error = INTcoinColors.Error,
    onError = Color.White,
    outline = INTcoinColors.Border,
    outlineVariant = INTcoinColors.Border.copy(alpha = 0.5f)
)

/**
 * INTcoin typography
 */
val INTcoinTypography = Typography(
    displayLarge = TextStyle(
        fontWeight = FontWeight.Bold,
        fontSize = 57.sp,
        lineHeight = 64.sp
    ),
    displayMedium = TextStyle(
        fontWeight = FontWeight.Bold,
        fontSize = 45.sp,
        lineHeight = 52.sp
    ),
    displaySmall = TextStyle(
        fontWeight = FontWeight.Bold,
        fontSize = 36.sp,
        lineHeight = 44.sp
    ),
    headlineLarge = TextStyle(
        fontWeight = FontWeight.Bold,
        fontSize = 32.sp,
        lineHeight = 40.sp
    ),
    headlineMedium = TextStyle(
        fontWeight = FontWeight.SemiBold,
        fontSize = 28.sp,
        lineHeight = 36.sp
    ),
    headlineSmall = TextStyle(
        fontWeight = FontWeight.SemiBold,
        fontSize = 24.sp,
        lineHeight = 32.sp
    ),
    titleLarge = TextStyle(
        fontWeight = FontWeight.SemiBold,
        fontSize = 22.sp,
        lineHeight = 28.sp
    ),
    titleMedium = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 16.sp,
        lineHeight = 24.sp
    ),
    titleSmall = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 14.sp,
        lineHeight = 20.sp
    ),
    bodyLarge = TextStyle(
        fontWeight = FontWeight.Normal,
        fontSize = 16.sp,
        lineHeight = 24.sp
    ),
    bodyMedium = TextStyle(
        fontWeight = FontWeight.Normal,
        fontSize = 14.sp,
        lineHeight = 20.sp
    ),
    bodySmall = TextStyle(
        fontWeight = FontWeight.Normal,
        fontSize = 12.sp,
        lineHeight = 16.sp
    ),
    labelLarge = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 14.sp,
        lineHeight = 20.sp
    ),
    labelMedium = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 12.sp,
        lineHeight = 16.sp
    ),
    labelSmall = TextStyle(
        fontWeight = FontWeight.Medium,
        fontSize = 11.sp,
        lineHeight = 16.sp
    )
)

/**
 * INTcoin shapes
 */
val INTcoinShapes = Shapes(
    extraSmall = RoundedCornerShape(4.dp),
    small = RoundedCornerShape(8.dp),
    medium = RoundedCornerShape(12.dp),
    large = RoundedCornerShape(16.dp),
    extraLarge = RoundedCornerShape(24.dp)
)

/**
 * INTcoin design constants
 */
object INTcoinDesign {
    val CornerRadiusSmall = 8.dp
    val CornerRadiusMedium = 12.dp
    val CornerRadiusLarge = 16.dp
    val CornerRadiusXL = 24.dp

    val PaddingSmall = 8.dp
    val PaddingMedium = 16.dp
    val PaddingLarge = 24.dp
    val PaddingXL = 32.dp

    val MinTouchTarget = 44.dp
    val ButtonHeight = 52.dp
    val IconSizeMedium = 24.dp
    val IconSizeLarge = 32.dp
    val IconSizeXL = 48.dp
}

/**
 * INTcoin theme composable
 */
@Composable
fun INTcoinTheme(
    content: @Composable () -> Unit
) {
    val colorScheme = INTcoinDarkColorScheme
    val view = LocalView.current

    if (!view.isInEditMode) {
        SideEffect {
            val window = (view.context as Activity).window
            window.statusBarColor = colorScheme.background.toArgb()
            window.navigationBarColor = colorScheme.background.toArgb()
            WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = false
            WindowCompat.getInsetsController(window, view).isAppearanceLightNavigationBars = false
        }
    }

    MaterialTheme(
        colorScheme = colorScheme,
        typography = INTcoinTypography,
        shapes = INTcoinShapes,
        content = content
    )
}
